### ISSUE TITLE
01-logistic-regression.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/marcoakai_machine_learning#3]
@Marcoakai 

---

### **Encoding: Factorize sin One-Hot**

Usas `pd.factorize()`como encoding del modelo. El factorize es una función de python válida para graficar y analizar nuestros datos. Pero a la hora del modelado los pasos son SPLIT  ----> ENCODING (por ejemplo one-hot-encoding, las variables factorizadas las dejamos fuera solo para graficar pero no para el modelo)  ----> ESCALING ( más info en el siguiente comentario)

```python
cat_cols = ["job","marital","education","default","housing","loan",...]
for col in cat_cols:
   df[col + "_n"] = pd.factorize(df[col])[0]
```

**Problema:** LogisticRegression interpreta `job=0, job=1, job=2` como si tuvieran orden/magnitud. No es un algoritmo muy vàlido para el escalado además de que tus datos están viendo 

---

### **ERROR CRÍTICO: Data Leakage en el escalado**

**Problema: Escalas TODO el dataset ANTES del split**

```python
scaler = MinMaxScaler()
df_scal = pd.DataFrame(scaler.fit_transform(df[num_variables]), ...)  # 
```

**¿Por qué es un error?**
El scaler "ve" los datos de test cuando calcula `fit_transform()` en todo el dataset. El modelo indirectamente tiene información del test set, lo que infla artificialmente las métricas.

**Corrección:** # 1. Split PRIMERO (sin escalar) ---> 2. ENCODING después de split ----> 3. Escalar DESPUÉS (solo fit en train) --->  4. SelectKBest con datos escalados si quieres hacerlo

**Comment by mezcolantriz** [comment]
@Marcoakai 
### Mejoras necesarias:

1. **No guardas encoder ni scaler:**
   ```python
   # Falta añadir:
   with open("../models/preprocessors.pkl", "wb") as f:
       pickle.dump({'encoder': encoder, 'scaler': scaler}, f)
   ```

2. **Falta visualización de la matriz de confusión:**
   ```python
   sns.heatmap(conf_matrix_opt, annot=True, fmt='d', cmap='Blues')
   ```

3. **`drop_duplicates()` sin `inplace=True`** - No tiene efecto

4. **Falta EDA visual** - Solo info() y value_counts(), no hay gráficas
